schema-to-pojo-org-json/src/test/java/org/sagebionetworks/schema/adapter/org/json/JSONObjectAdapterImplTest.java

```diff
@@ -131,6 +131,38 @@ public void testDoubleRoundTrip() throws JSONObjectAdapterException{
 	}
 	
 	@Test
+	public void testDoubleNaNRoundTrip() throws JSONObjectAdapterException{
+		double value = Double.NaN;
+		adapter.put(propertyKey, value);
+		assertTrue(adapter.has(propertyKey));
+		assertTrue(Double.isNaN(adapter.getDouble(propertyKey)));
+		// Make sure we can also get it as an object
+		assertTrue(Double.isNaN(Double.parseDouble((String)adapter.get(propertyKey))));
+	}
+	
+	@Test
+	public void testDoubleInfinityRoundTrip() throws JSONObjectAdapterException{
+		double value = Double.POSITIVE_INFINITY;
+		adapter.put(propertyKey, value);
+		assertTrue(adapter.has(propertyKey));
+		assertTrue(Double.isInfinite(adapter.getDouble(propertyKey)));
+		assertEquals(Double.POSITIVE_INFINITY, adapter.getDouble(propertyKey), 0d);
+		// Make sure we can also get it as an object
+		assertTrue(Double.isInfinite(Double.parseDouble((String)adapter.get(propertyKey))));
+	}
+	
+	@Test
+	public void testDoubleNegativeInfinityRoundTrip() throws JSONObjectAdapterException{
+		double value = Double.NEGATIVE_INFINITY;
+		adapter.put(propertyKey, value);
+		assertTrue(adapter.has(propertyKey));
+		assertTrue(Double.isInfinite(adapter.getDouble(propertyKey)));
+		assertEquals(Double.NEGATIVE_INFINITY, adapter.getDouble(propertyKey), 0d);
+		// Make sure we can also get it as an object
+		assertTrue(Double.isInfinite(Double.parseDouble((String)adapter.get(propertyKey))));
+	}
+	
+	@Test

```

Comment:
See: http://tools.ietf.org/html/rfc4627
"Numeric values that cannot be represented as sequences of digits
   (such as Infinity and NaN) are not permitted."
Are we sure this fix will actually work in a round trip to JSON?  Even if does will it break the R or Python client?